src/dataregistry/registrar.py

```diff
@@ -270,41 +306,63 @@ def register_dataset(self, relative_path, version,
         is_overwritable : bool, optional
             True if dataset may be overwritten (defaults to False).
 
-            Production datasets cannot be overwritten.
+            Note production datasets cannot be overwritten.
         old_location : str, optional
-            Absolute location of dataset to copy.
+            Absolute location of dataset to copy into the data registry.
 
-            If None dataset should already be at correct relative_path.
+            If None, dataset should already be at correct relative_path within
+            the data registry.
         copy : bool, optional
-            If true copy data from ``old_location`` to the database.
-            If False create a symlink (defaults to True).
-        is_dummy : bool
+            True to copy data from ``old_location`` into the data registry
+            (default behaviour).  
+            False to create a symlink.
+        is_dummy : bool, optional
             True for "dummy" datasets (no data is copied, for testing purposes
             only)
-        verbose : bool
+        verbose : bool, optional
             Provide some additional output information
+        owner : str, optional
+            Owner of the dataset. If None, defaults to $USER
+        owner_type : str, optional
+            Owner type: "user" (default), "group", or "production"
 
         Returns
         -------
         prim_key : int
             The dataset ID of the new row relating to this entry (else None)
         """
 
-        if (self._owner_type == "production"):
+        # Make sure the owner_type is legal
```

Comment:
All the code having to do with setting and verifying values for owner and owner_type should be moved to ```Registrar.__init__```